pynest/examples/Potjans_2014/network.py

```diff
@@ -0,0 +1,477 @@
+                        'delay': {
+                            'distribution': 'normal_clipped',
+                            'mu': mean_delays[i][j], 'sigma': std_delays[i][j],
+                            'low': self.sim_resolution
+                            }
+                        }
+                    if weight < 0:
+                        syn_dict['weight']['high'] = 0.0
+                    else:
+                        syn_dict['weight']['low'] = 0.0
+                    nest.Connect(
+                        source_pop, target_pop,
+                        conn_spec=conn_dict_rec,
+                        syn_spec=syn_dict
+                        )
+
+    def connect_poisson(self):
+        """ Connects the Poisson generators to the microcircuit."""
+        print('Poisson background input is connected')
+        for i, target_pop in enumerate(self.pops):
+            conn_dict_poisson = {'rule': 'all_to_all'}
+            syn_dict_poisson = {
+                'model': 'static_synapse',
+                'weight': self.w_ext,
+                'delay': self.net_dict['poisson_delay']
+                }
+            nest.Connect(
+                self.poisson[i], target_pop,
+                conn_spec=conn_dict_poisson,
+                syn_spec=syn_dict_poisson
+                )
+
+    def connect_thalamus(self):
+        """ Connects the thalamic population to the microcircuit."""
+        print('Thalamus connection established')
+        for i, target_pop in enumerate(self.pops):
+            conn_dict_th = {
+                'rule': 'fixed_total_number',
+                'N': int(self.nr_synapses_th[i])
+                }
+            syn_dict_th = {
+                'weight': {
+                    'distribution': 'normal_clipped',
+                    'mu': self.thalamic_weight,
+                    'sigma': (
+                        self.thalamic_weight * self.net_dict['PSP_sd']
+                        ),
+                    'low': 0.0
+                    },
+                'delay': {
+                    'distribution': 'normal_clipped',
+                    'mu': self.stim_dict['delay_th'][i],
+                    'sigma': self.stim_dict['delay_th_sd'][i],
+                    'low': self.sim_resolution
+                    }
+                }
+            nest.Connect(
+                self.thalamic_population, target_pop,
+                conn_spec=conn_dict_th, syn_spec=syn_dict_th
+                )
+
+    def connect_dc_generator(self):
+        """ This function connects the DC generator to the microcircuit."""
+        print('DC Generator connection established')
+        for i, target_pop in enumerate(self.pops):
+            if self.stim_dict['dc_input']:
+                nest.Connect(self.dc[i], target_pop)
+
+    def connect_devices(self):
+        """ Connects the recording devices to the microcircuit."""
+        print('%s of 2 Devices connected' % (len(self.net_dict['rec_dev'])))
```

Comment:
It prints "1 of 2 Devices connected" if only a spike detector is used which sounds as if something was missing. I suggest to be specific and print "Spike Detector connected" or/and "Voltmeter connected".